lib/Doctrine/ORM/Query/TreeWalker.php

```diff
@@ -16,12 +16,10 @@ interface TreeWalker
     /**
      * Initializes TreeWalker with important information about the ASTs to be walked.
      *
-     * @param AbstractQuery $query           The parsed Query.
-     * @param ParserResult  $parserResult    The result of the parsing process.
-     * @param mixed[]       $queryComponents The query components (symbol table).
+     * @param mixed[] $queryComponents The query components (symbol table).
      * @psalm-param array<string, QueryComponent> $queryComponents The query components (symbol table).
      */
-    public function __construct($query, $parserResult, array $queryComponents);
+    public function __construct(AbstractQuery $query, ParserResult $parserResult, array $queryComponents);
```

Comment:
This forces us to use `AbstractQuery` in classes that implement the interface. We can still annotate the parameter with `Query`, but this is kind of weird.